service/history/ndc/history_replicator.go

```diff
@@ -297,6 +297,10 @@ func (r *HistoryReplicatorImpl) doApplyEvents(
 					metrics.OperationTag(metrics.ReplicateHistoryEventsScope))
 				return nil
 			}
+			err = task.skipDuplicatedEvents(prepareHistoryBranchOut.EventsApplyIndex)
+			if err != nil {
+				return err
+			}
 
 			mutableState, isRebuilt, err := r.mutableStateMapper.GetOrRebuildCurrentMutableState(
```

Comment:
Could skip those task inside this function?

Reply:
My intention is to immediately skip all duplicated events before doing any further actions. Let me know if you have strong opinion on this.